src/net/rpc.rs

```diff
@@ -1509,6 +1510,86 @@ impl ConversationHttp {
         return response.send(http, fd).map(|_| ());
     }
 
+    /// Handle a GET health.
+    /// If the node's height is not up to the max height of all of its bootstrap nodes, we return
+    /// false for the health check.
+    /// Additionally return what percent of the max height the node height is at.
+    fn handle_get_health<W: Write>(
+        http: &mut StacksHttp,
+        fd: &mut W,
+        req: &HttpRequestType,
+        network_id: u32,
+        sortdb: &SortitionDB,
+        chainstate: &StacksChainState,
+        peerdb: &PeerDB,
+        block_stats: Option<&HashMap<NeighborKey, NeighborBlockStats>>,
+    ) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+
+        let block_height = match chainstate.get_stacks_chain_tip(sortdb)? {
+            Some(tip) => tip.height,
+            None => 0,
+        };
+
+        let initial_neighbors =
+            PeerDB::get_valid_initial_neighbors(peerdb.conn(), network_id, block_height).unwrap();
+
+        if initial_neighbors.len() == 0 {
+            // no bootstrap nodes found, unable to determine health.
+            let response = HttpResponseType::NotFound(
```

Comment:
I'm not sure if this should be an "error". Maybe it can be a valid GetHealthResponse, that simply indicates things are bad.

That way:
1) This function can have a simpler definition.
1) The client can expect a standard form, especially if they want to write code to parse the response.

What are your thoughts on that?

Reply:
Yes I see. I'll make a decision on this after discussing what the returns should look like with devops.

Reply:
i agree with @gregorycoppola here, but i don't have an opinion on how to represent this data with the returned json. On the other hand, if the node didn't have any neighbors to sync with...that would be a larger problem for the node to deal with anyways. 

perhaps in this case, the output could be returned like this:
```
{
  "is_healthy": false,
  "percent_of_blocks_synced":  null
}
```
or some other identifier that signals the issue is with a lack of neighbors

Reply:
I guess, in this case, that the "health" endpoint has the responsibility to make a binary decision on healthiness.

I was thinking maybe the client would want to weigh several factors and decide on healthiness. But, I guess now after learning about the context, that I guess the endpoint does need to weight whatever factors and make a decision itself.

Reply:
If I return type HttpResponseType:::GetHealth, the status code will be 200, which doesn't seem like the desired behavior here. 

I could make a specific response type HttpResponseType:::GetHealthError that has a 400 status code. This will deviate from how we "fail" in the rest of our rpc calls, but I think that's fine because we are trying to use this endpoint with Kubernetes probes.

Reply:
But we can still return a lot of info in the response in case we want to understand why the health check failed.

Reply:
in this case though, we're only concerned with the "healthiness" of the node itself at a specific moment in time. 
I would say that the node making those binary decisions *about itself* are fine. 
@pavitthrap yes - if you wanted to convey specific reasons for a non-healthy node (or even why it's healthy), more k:v pairs and differing HTTP response codes would be the best way to accomplish this i would think. 
My only ask if we do add more and more response codes is that we document what each code means in the context of this endpoint.